src/components/admin.tsx

```diff
@@ -761,14 +761,14 @@ function Admin({ user }: adminProps) {
     if (navigator.share) {
       setIsSettingAssignLink(true);
       try {
-        await setTimedLink(linktype, postalcode, linkId, hours);
         navigator
           .share({
             title: title,
             text: body,
             url: url
           })
-          .then((_) => {
+          .then(async (_) => {
```

Comment:
Thanks bro for the work while I am away.

`navigator.share` will return a promise.

So I think you can make this part more cleaner by awaiting the share promise instead of .then followed by another async function declaration.

For eg,

```
await navigator.share(....);
await settimeLink(....);
```

Since the functions are within a try/catch, any cancellation of sharing will be handled.
If navigator.share succeeds, it will run the next line of code.